src/components/screens/layout-screen-html.tsx

```diff
@@ -0,0 +1,233 @@
+import { useState } from "react";
+
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { ReduxActions, ReduxState } from "../../store";
+import { setSelectedLayout, setLayouts } from "../../actions/layouts";
+import { History } from "history";
+import styles from "../../styles/components/layout-screen/layout-editor-view";
+import {
+  CircularProgress,
+  TextField,
+  Select,
+  MenuItem,
+  InputLabel,
+  FormControl,
+  SelectChangeEvent,
+} from "@mui/material";
+import { WithStyles } from '@mui/styles';
+import withStyles from '@mui/styles/withStyles';
+import { KeycloakInstance } from "keycloak-js";
+import { PageLayout, Exhibition, DeviceModel, ScreenOrientation, SubLayout } from "../../generated/client";
+import BasicLayout from "../layouts/basic-layout";
+import ElementNavigationPane from "../layouts/element-navigation-pane";
+import EditorView from "../editor/editor-view";
+import { AccessToken, ActionButton } from '../../types';
+import strings from "../../localization/strings";
+import theme from "../../styles/theme";
+
+/**
+ * Component props
+ */
+interface Props extends WithStyles<typeof styles> {
+  history: History;
+  keycloak: KeycloakInstance;
+  accessToken: AccessToken;
+  exhibitions: Exhibition[];
+  deviceModels: DeviceModel[];
+  layoutId: string;
+  layout?: PageLayout;
+  layouts: PageLayout[];
+  subLayouts: SubLayout[];
+  setLayouts: typeof setLayouts;
+  setSelectedLayout: typeof setSelectedLayout;
+}
+
+/**
+ * Component for html layout screen
+ */
+export const LayoutScreenHTML: React.FC<Props> = ({
+  history,
+  keycloak,
+  deviceModels,
+  layout,
+  classes
+}) => {
+  const [ name, setName ] = useState("");
+  const [ deviceModelId, setDeviceModelId ] = useState("");
+  const [ screenOrientation, setScreenOrientation ] = useState<ScreenOrientation>(ScreenOrientation.Portrait);
+  const [ view, setView ] = useState("VISUAL")
+  const [ dataChanged, setDataChanged ] = useState(false);
+
```

Comment:
Add functionality for fetching the layout from the API.
Currently if you reload this page, the selected layout is lost from state and this screen is left in an endless-loop.
Also navigate back to listing if the layout couldn't be found.